src/components/base/Modal/index.js

```diff
@@ -28,11 +28,20 @@ const springConfig = {
 
 const mapStateToProps: Function = (
   state,
-  { name, isOpened }: { name: string, isOpened?: boolean },
-): Object => ({
-  isOpened: isOpened || (name && isModalOpened(state, name)),
-  data: getModalData(state, name),
-})
+  { name, isOpened, onBeforeOpen }: { name: string, isOpened?: boolean, onBeforeOpen: Function },
+): Object => {
```

Comment:
`*` would likely help more than `Object` because it would likely infer correctly the type